### ISSUE TITLE
Building Ubuntu 22.04 Image Fails on google-cloud-sdk.sh

### ISSUE THREAD

**Issue by v1ferrarij** [actions/runner-images#7200]
### Description

I am building an Image using Packer, I have used the **main** branch and release **ubuntu22/20230219.1** both are failing with the same error:

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

ubuntu22/20230219.1

### Is it regression?

N/A

### Expected behavior

The build to succeed with no errors.

### Actual behavior

The build is failing with no real descriptive error:

==> azure-arm.build_vhd: Provisioning with shell script: images/linux/scripts/installers/google-cloud-sdk.sh
==> azure-arm.build_vhd: Provisioning step had errors: Running the cleanup provisioner, if present...
==> azure-arm.build_vhd:
==> azure-arm.build_vhd: Deleting individual resources ...
==> azure-arm.build_vhd: Adding to deletion queue -> Microsoft.Compute/virtualMachines : 'pkrvmm8u824qniw'
==> azure-arm.build_vhd: Adding to deletion queue -> Microsoft.Network/networkInterfaces : 'pkrnim8u824qniw'

### Repro steps

1. Checkout to **main** or **ubuntu22/20230219.1.**
2. Amend source azure-arm with relevant parameters.
3. run **packer build .\images\linux\ubuntu2204.pkr.hcl**

**Comment by ddobranic** [comment]
Hello @v1ferrarij, we will investigate. Thank you for reporting.

**Comment by ddobranic** [comment]
@v1ferrarij I cannot reproduce the issue. Image is building successfully from our repository main branch. Have you tried using latest code?

**Comment by v1ferrarij** [comment]
I have re-cloned the repo and run again from **main**, still the same error. 

The only thing I changed in the entire repo was this file **ubuntu2204.pkr.hcl** replacing the default "azure-arm" "build_vhd" ... with my own below, everything else is exactly the same:

```hcl
source "azure-arm" "build_vhd" {
  build_resource_group_name = "devops-rg"

  image_offer     = "0001-com-ubuntu-server-jammy"
  image_publisher = "canonical"
  image_sku       = "22_04-lts"
  location        = var.location
  os_disk_size_gb = "86"
  os_type         = "Linux"
  vm_size         = var.vm_size

  managed_image_name                = "ubuntu2204-devops"
  managed_image_resource_group_name = "devops-rg"

  use_azure_cli_auth = true

  shared_image_gallery_destination {
    subscription   = var.subscription_id
    resource_group = "devops-rg"
    gallery_name   = "devops-cg"
    image_version  = "1.0.0"
    image_name     = "ubuntu2204"
  }
}
```

**Comment by mikhailkoliada** [comment]
Сould you please post a full build log?

**Comment by erik-de-bont** [comment]
I get the same error with a Ubuntu 20.04 build.

**Comment by panekj** [comment]
The build error is happening due to intermittent issues with the https://packages.cloud.google.com since yesterday 7am (26/02)
Additional context: https://twitter.com/dims/status/1630270940855644165?s=20

(you won't see much in the log)
<details>
<summary>Log</summary>

```
==> act.docker.base_image: Provisioning with shell script: ./scripts/installers/google-cloud-sdk.sh
==> act.docker.base_image: + REPO_URL=https://packages.cloud.google.com/apt
==> act.docker.base_image: + echo 'deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main'
==> act.docker.base_image: + curl -sL https://packages.cloud.google.com/apt/doc/apt-key.gpg
==> act.docker.base_image: + apt-get update -y
    act.docker.base_image: Hit:1 https://ppa.launchpadcontent.net/ubuntu-toolchain-r/test/ubuntu jammy InRelease
    act.docker.base_image: Hit:2 https://packages.microsoft.com/ubuntu/22.04/prod jammy InRelease
    act.docker.base_image: Hit:3 http://security.ubuntu.com/ubuntu jammy-security InRelease
    act.docker.base_image: Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease
    act.docker.base_image: Hit:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
    act.docker.base_image: Hit:6 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
    act.docker.base_image: Get:7 https://packages.cloud.google.com/apt cloud-sdk InRelease [6361 B]
    act.docker.base_image: Err:7 https://packages.cloud.google.com/apt cloud-sdk InRelease
    act.docker.base_image:   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
    act.docker.base_image: Reading package lists...
==> act.docker.base_image: W: GPG error: https://packages.cloud.google.com/apt cloud-sdk InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
==> act.docker.base_image: E: The repository 'https://packages.cloud.google.com/apt cloud-sdk InRelease' is not signed.
==> act.docker.base_image: + apt-get install -y google-cloud-sdk
    act.docker.base_image: Reading package lists...
    act.docker.base_image: Building dependency tree...
==> act.docker.base_image: E: Unable to locate package google-cloud-sdk
==> act.docker.base_image: + rm /etc/apt/sources.list.d/google-cloud-sdk.list
==> act.docker.base_image: + rm /usr/share/keyrings/cloud.google.gpg
    act.docker.base_image: Reading state information...
==> act.docker.base_image: + echo 'google-cloud-sdk https://packages.cloud.google.com/apt'
==> act.docker.base_image: + invoke_tests CLI.Tools 'Google Cloud SDK'
    act.docker.base_image: Pester v5.4.0
    act.docker.base_image:
    act.docker.base_image: Starting discovery in 1 files.
    act.docker.base_image: Discovery found 11 tests in 110ms.
    act.docker.base_image: Filter 'FullName' set to ('Google Cloud SDK').
    act.docker.base_image: Filters selected 1 tests to run.
    act.docker.base_image: Running tests.
    act.docker.base_image:
    act.docker.base_image: Running tests from '/imagegeneration/tests/CLI.Tools.Tests.ps1'
    act.docker.base_image: Describing Google Cloud SDK
    act.docker.base_image:   [-] Google Cloud SDK 71ms (53ms|19ms)
    act.docker.base_image:    Command 'gcloud --version' has finished with exit code
    act.docker.base_image:        /usr/bin/bash: line 1: gcloud: command not found
    act.docker.base_image:    at "gcloud --version" | Should -ReturnZeroExitCode, /imagegeneration/tests/CLI.Tools.Tests.ps1:46
    act.docker.base_image:    at <ScriptBlock>, /imagegeneration/tests/CLI.Tools.Tests.ps1:46
    act.docker.base_image: Tests completed in 451ms
    act.docker.base_image: Tests Passed: 0, Failed: 1, Skipped: 0 NotRun: 10
    act.docker.base_image:
    act.docker.base_image: Containers            : {[-] /imagegeneration/tests/CLI.Tools.Tests.ps1}
    act.docker.base_image: Result                : Failed
    act.docker.base_image: FailedCount           : 1
    act.docker.base_image: FailedBlocksCount     : 0
    act.docker.base_image: FailedContainersCount : 0
    act.docker.base_image: PassedCount           : 0
    act.docker.base_image: SkippedCount          : 0
    act.docker.base_image: NotRunCount           : 10
    act.docker.base_image: TotalCount            : 11
    act.docker.base_image: Duration              : 00:00:00.4512042
    act.docker.base_image: Executed              : True
==> act.docker.base_image: Exception: Test run has failed
==> act.docker.base_image: 
    act.docker.base_image: ExecutedAt            : 02/26/2023 07:23:16
    act.docker.base_image: Version               : 5.4.0
    act.docker.base_image: PSVersion             : 7.2.10
    act.docker.base_image: PSBoundParameters     : {[Configuration, PesterConfiguration]}
    act.docker.base_image: Plugins               : 
    act.docker.base_image: PluginConfiguration   : 
    act.docker.base_image: PluginData            : 
    act.docker.base_image: Configuration         : PesterConfiguration
    act.docker.base_image: DiscoveryDuration     : 00:00:00.1011825
    act.docker.base_image: UserDuration          : 00:00:00.0575691
    act.docker.base_image: FrameworkDuration     : 00:00:00.2924526
    act.docker.base_image: Failed                : {[-] Google Cloud SDK}
    act.docker.base_image: FailedBlocks          : {}
    act.docker.base_image: FailedContainers      : {}
    act.docker.base_image: Passed                : {}
    act.docker.base_image: Skipped               : {}
    act.docker.base_image: NotRun                : {[ ] Azure CLI, [ ] az devops, [ ] Aliyun CLI, [ ] AWS
    act.docker.base_image:                         CLI…}
    act.docker.base_image: Tests                 : {[ ] Azure CLI, [ ] az devops, [ ] Aliyun CLI, [ ] AWS
    act.docker.base_image:                         CLI…}
    act.docker.base_image: CodeCoverage          : 
    act.docker.base_image:
==> act.docker.base_image: Provisioning step had errors: Running the cleanup provisioner, if present...
==> act.docker.base_image: Killing the container: b4cdee35ea2a979b9d95fc6d2d3ca938b8828a9d3056fdb6e0f5a48c1d651de2
Build 'act.docker.base_image' errored after 20 minutes 54 seconds: Script exited with non-zero exit status: 1. Allowed exit codes are: [0]
```

</details>

**Comment by mikhailkoliada** [comment]
I have seen the error like 12h ago too :) It was due to apt not being able to verify a gpg signature of the google's repository because the gpg binary pubkey (https://packages.cloud.google.com/apt/doc/apt-key.gpg) was not accessible, I see it now all works as expected though, so you have to experience problems no more :)
